app/jobs/badge_assignment_check_job.rb

```diff
@@ -0,0 +1,32 @@
+class BadgeAssignmentCheckJob < ApplicationJob
+  queue_as :default
+
+  def perform(days_to_check: 31, programmes_to_check: [])
+    missing_badges = []
+    programmes_to_check.each do |programme|
```

Comment:
Minor code style note: If you were to use [`.each_with_object`](https://ruby-doc.org/3.1.4/Enumerable.html#method-i-each_with_object) here you wouldn't need to define `missing_badges`, or reference it at the end of the method. There's no functional difference though.